custom/intrahealth/sqldata.py

```diff
@@ -4149,26 +4156,26 @@ def clean_rows(data_to_clean):
 
             return stocks_list_to_return
 
+        fresh_records_dict = {}
         for row in rows:
-            if not rows_to_return:
-                rows_to_return.append(row)
+            pps_id = row['pps_id']
+            product_id = row['product_id']
+            if pps_id not in fresh_records_dict.keys():
```

Comment:
nit: there's no need to call `.keys()` when checking whether a key is in a dictionary.
```suggestion
            if pps_id not in fresh_records_dict:
```